stripe/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsViewModel.kt

```diff
@@ -52,29 +63,61 @@ internal class PaymentOptionsViewModel(
 
     fun onUserSelection() {
         selection.value?.let { paymentSelection ->
+            // TODO: Should the payment selection in the event be the saved or new item?
             eventReporter.onSelectPaymentOption(paymentSelection)
-            prefsRepository.savePaymentSelection(paymentSelection)
-            processSelection(paymentSelection)
+
+            val requestSaveNewCard =
+                (paymentSelection as? PaymentSelection.New)?.shouldSavePaymentMethod
+                    ?: false
+            if (requestSaveNewCard) {
+                processSaveNewCard(paymentSelection)
+            } else {
+                processUnsavedNewCard(paymentSelection)
+            }
         }
     }
 
-    private fun processSelection(paymentSelection: PaymentSelection) {
-        val requestSaveNewCard =
-            (paymentSelection as? PaymentSelection.New)?.shouldSavePaymentMethod
-                ?: false
-
-        if (requestSaveNewCard) {
-            // TODO: Update the returned value with the savedCard rather than the NewCard
-            // so that we don't jump the next time.
-            _viewState.value = ViewState.PaymentOptions.FinishProcessing {
-                _viewState.value = ViewState.PaymentOptions.ProcessResult(
-                    PaymentOptionResult.Succeeded(paymentSelection)
-                )
+    private fun processUnsavedNewCard(paymentSelection: PaymentSelection) {
+        _viewState.value = ViewState.PaymentOptions.Ready
+        prefsRepository.savePaymentSelection(paymentSelection)
```

Comment:
Should we add `StartProcessing` and `FinishProcessing` since this is technically an async call?

Reply:
Are you thinking have the finish processing happen when the save to the preference repository is complete? :D

Reply:
Yeah, but maybe this is overkill + would result in weird animations

Reply:
We will wait on it, if it is an issue we can address it then.